### PR TITLE
Adding functionality to execute a callback on every (x) ticks.

### PR DESCRIPTION
**Pull request by barrychapman**
The callback is configurable, as is the number of ticks to execute callback.

Will allow the user to specify a callback to run every time a tick is processed (however many ms that may be).

So - if the user wanted to update a progress bar - they could do that in the step callback, and then it would stop once the timer completes. Better than having a separate function for it.

### CODE REVIEW

File: jquery.time-to.js [hunk 1]
```diff
@@ -401,6 +410,12 @@
         $li.eq(1).html(n);
         n += step;
 
+        if (typeof this.data('step') === 'function') {
+            if (this.data('tickCount') % this.data('stepCount') === 0) {
```
Comment (Lexxus) [inline jquery.time-to.js]: `if (typeof data.step === 'function' && tc % data.stepCount === 0) {`

File: jquery.time-to.js [hunk 2]
```diff
@@ -401,6 +410,12 @@
...
+                this.data('step')();
```
Comment (Lexxus) [inline jquery.time-to.js]: Add `this.data('tickCount', 0);` to prevent number overload.

File: README.md [hunk 3]
```diff
@@ -58,6 +58,8 @@ Object with initial settings:
 
 * **callback**:   *function* that call when countdown end, default **null**;
```
Comment (Lexxus) [inline README.md]: > that is called

sorry for my bad English )

Comment (barrychapman) [inline README.md]: That is called once - when the timer is complete.

What I have added, will execute while the timer is running. If they specify the stepCount to be '5', then the callback will execute every 5 ticks.

File: jquery.time-to.js (hunk 2 above)
Comment (barrychapman) [inline jquery.time-to.js]: Can you clarify what you mean here?

File: README.md (hunk 3 above)
Comment (barrychapman) [inline README.md]: As per your application, the time duration of a tick is completely configurable. If a user specifies a tick to be 1500ms, then a 3 step timer callback will execute every 4500ms.

Comment (Lexxus) [inline README.md]: Yes, I know. I just figured out the grammar mistake here.

File: jquery.time-to.js (hunk 2 above)
Comment (Lexxus) [inline jquery.time-to.js]: When `tickCount` reach MAX_SAFE_INTEGER further increasing won't change a value.
So I offer to reset this variable each time.

File: jquery.time-to.js [hunk 4]
```diff
@@ -393,6 +398,10 @@
             digit = data.iSec;
         }
 
+        tc = data.tickCount + 1;
+
+        this.data('tickCount', tc);
```
Comment (Lexxus) [inline jquery.time-to.js]: If we don't use `tc` anymore join these two lines to
`this.data('tickCount', data.tickCount + 1);`
And remove `tc` in declaration block.

**Pull request merged**